src/testRunner/unittests/tsbuild/resolveJsonModule.ts

```diff
@@ -53,7 +53,12 @@ namespace ts {
 
 export default hello.hello`);
             const allExpectedOutputs = ["/src/dist/src/index.js", "/src/dist/src/index.d.ts", "/src/dist/src/index.json"];
-            verifyProjectWithResolveJsonModuleWithFs(fs, "/src/tsconfig_withIncludeOfJson.json", allExpectedOutputs);
+            verifyProjectWithResolveJsonModuleWithFs(
+                fs,
+                "/src/tsconfig_withIncludeOfJson.json",
+                allExpectedOutputs,
+                errorDiagnostic([Diagnostics.Cannot_write_file_0_because_it_would_be_overwritten_by_multiple_input_files, "/src/dist/src/index.d.ts"])

```

Comment:
I think for the files where .json file will not be written we should also skip .d.ts write for that file.. Because if its written at same location in most config files `.d.ts` takes preference over `.json` but module resolution picks up `.json` file thus we now get both files?
```
c:\temp\test4>dir
 Volume in drive C is OSDisk
 Volume Serial Number is F41A-DBC9

 Directory of c:\temp\test4

09/25/2019  08:37 AM    <DIR>          .
09/25/2019  08:37 AM    <DIR>          ..
09/25/2019  08:32 AM                43 a.js
09/25/2019  08:31 AM                31 a.ts
09/25/2019  08:31 AM                 0 b.d.ts
09/25/2019  08:30 AM                 2 b.json
09/25/2019  08:32 AM                72 tsconfig.json
               5 File(s)            148 bytes
               2 Dir(s)  798,188,781,568 bytes free

c:\temp\test4>node c:\TypeScript\built\local\tsc.js --listFiles
c:/TypeScript/built/local/lib.d.ts
c:/TypeScript/built/local/lib.es5.d.ts
c:/TypeScript/built/local/lib.dom.d.ts
c:/TypeScript/built/local/lib.webworker.importscripts.d.ts
c:/TypeScript/built/local/lib.scripthost.d.ts
c:/temp/test4/b.json
c:/temp/test4/a.ts
c:/temp/test4/b.d.ts
```

Reply:
What do you mean by "we now get both"? An import is only going to resolve to one of them.

Reply:
Yes import is going to resolve to `b.json` in this case but b.d.ts which is output of `b.json` will also be part of the project. So my program has more files after I compile for the first time? So question is do we emit `b.json.d.ts` instead of `b.d.ts` or not emit b.d.ts if json file is not going to be emitted?

Reply:
We definitely emit the declaration and don't issue an error - it'd be super annoying if `require`ing your own `package.json` broke incremental for your project. Still, I'm not getting it - once the declaration file is emitted, the sibling files are going to resolve to either the original json, _or_ the declaration file (not both). So how do both even get included in the project? It's not materially different than emitting a `.d.ts` alongside a `.ts` file, imo.